test/vfd_swmr_group_writer.c

```diff
@@ -112,216 +157,2736 @@ state_init(state_t *s, int argc, char **argv)
                     goto error;
                 }
 
-                if (ch == 'a')
-                    s->asteps = (unsigned)tmp;
-                else if (ch == 'c')
-                    s->csteps = (unsigned)tmp;
-                else if (ch == 'n')
-                    s->nsteps = (unsigned)tmp;
-                else if (ch == 'u')
-                    s->update_interval = (unsigned)tmp;
-                break;
-            case 'b':
-                s->filetype = H5T_STD_U32BE;
-                break;
-            case 'N':
-                s->use_named_pipes = false;
-                break;
-            case 'q':
-                verbosity = 0;
-                break;
-            case '?':
-            default:
-                usage(s->progname);
-                break;
+                if (ch == 'a')
+                    s->asteps = (unsigned)tmp;
+                else if (ch == 'c')
+                    s->csteps = (unsigned)tmp;
+                else if (ch == 'n')
+                    s->nsteps = (unsigned)tmp;
+                else if (ch == 'u')
+                    s->update_interval = (unsigned)tmp;
+                break;
+            case 'b':
+                s->filetype = H5T_STD_U32BE;
+                break;
+            case 'N':
+                s->use_named_pipes = false;
+                break;
+            case 'A':
+                if (strcmp(optarg, "compact") == 0)
+                    s->at_pattern = 'c';
+                else if (strcmp(optarg, "dense") == 0)
+                    s->at_pattern = 'd';
+                else if (strcmp(optarg, "compact-add-to-dense") == 0) 
+                    s->at_pattern = 't';
+                else if (strcmp(optarg, "compact-del") == 0) 
+                    s->at_pattern = 'C';
+                else if (strcmp(optarg, "dense-del") == 0) 
+                    s->at_pattern = 'D';
+                else if (strcmp(optarg, "dense-del-to-compact") == 0) 
+                    s->at_pattern = 'T';
+                else if (strcmp(optarg, "modify") == 0) 
+                    s->at_pattern = 'M';
+                else if (strcmp(optarg,"add-vstr") ==0)
+                    s->at_pattern = 'v';
+                else if (strcmp(optarg, "remove-vstr") == 0) 
+                    s->at_pattern = 'r';
+                else if (strcmp(optarg, "modify-vstr") == 0) 
+                    s->at_pattern = 'm';
+                else {
+                    H5_FAILED(); AT();
+                    printf("Invalid -A argument \"%s\"", optarg);
+                    goto error;
+                }
+                break;
+            case 'q':
+                verbosity = 0;
+                break;
+            case '?':
+            default:
+                usage(s->progname);
+                break;
+        }
+    }
+    argc -= optind;
+    argv += optind;
+
+    /* space for attributes */
+    if ((s->one_by_one_sid = H5Screate_simple(1, &dims, &dims)) < 0) {
```

Comment:
I don't see where this ID is being closed.

Reply:
This is used by add_attr(). Around line 433, ` hid_t sid = s->one_by_one_sid;
`

Reply:
That's fine.  But that ID should be closed in the end.